www-apps/redmine/redmine-3.4.12.ebuild

```diff
@@ -0,0 +1,232 @@
+
+	# bug #406605
+	rm .{git,hg}ignore || die
+
+	cat > "${T}/50${PN}" <<-EOF || die
+		CONFIG_PROTECT="${EROOT%/}${REDMINE_DIR}/config"
+		CONFIG_PROTECT_MASK="${EROOT%/}${REDMINE_DIR}/config/locales ${EROOT%/}${REDMINE_DIR}/config/settings.yml"
+	EOF
+
+	# remove ldap staff module if disabled to avoid #413779
+	use ldap || rm app/models/auth_source_ldap.rb || die
+
+	# Fixing versions in Gemfile
+	eapply "${FILESDIR}/${P}_gemfile_versions.patch"
+
+	sed -i -e "/group :development do/,/end$/d" Gemfile || die
+	sed -i -e "/group :test do/,/end$/d" Gemfile || die
+
+	if ! use imagemagick ; then
+		sed -i -e "/group :rmagick do/,/end$/d" Gemfile || die
+	fi
+	if ! use ldap ; then
+		sed -i -e "/group :ldap do/,/end$/d" Gemfile || die
+	fi
+	if ! use markdown ; then
+		sed -i -e "/group :markdown do/,/end$/d" Gemfile || die
+	fi
+}
+
+all_ruby_install() {
+	dodoc doc/* README.rdoc
+	rm -r doc appveyor.yml CONTRIBUTING.md README.rdoc || die
+
+	keepdir /var/log/${PN}
+
+	# Additional dependency for Gemfile (#657156)
+	if use fastcgi; then
+		echo "gem \"fcgi\"" > Gemfile.local
+	fi
+
+	insinto "${REDMINE_DIR}"
+	doins -r .
+	insinto "${REDMINE_DIR}/config"
+	doins "${FILESDIR}/additional_environment.rb"
+	keepdir "${REDMINE_DIR}/files"
+	keepdir "${REDMINE_DIR}/public/plugin_assets"
+
+	fowners -R redmine:redmine \
+		"${REDMINE_DIR}/config" \
+		"${REDMINE_DIR}/files" \
+		"${REDMINE_DIR}/public/plugin_assets" \
+		"${REDMINE_DIR}/tmp" \
+		/var/log/${PN}
+
+	fowners redmine:redmine "${REDMINE_DIR}"
+
+	# protect sensitive data, see bug #406605
+	fperms -R go-rwx \
+		"${REDMINE_DIR}/config" \
+		"${REDMINE_DIR}/files" \
+		"${REDMINE_DIR}/tmp" \
+		/var/log/${PN}
+
+	if use passenger; then
+		has_apache
+		insinto "${APACHE_VHOSTS_CONFDIR}"
+		doins "${FILESDIR}/10_redmine_vhost.conf"
+	fi
+
+	# logrotate
+	insinto /etc/logrotate.d
+	newins "${FILESDIR}"/redmine.logrotate redmine
+
+	newconfd "${FILESDIR}/${PN}.confd" ${PN}
+	newinitd "${FILESDIR}/${PN}-3.initd" ${PN}
+	doenvd "${T}/50${PN}"
+}
+
+pkg_postinst() {
+	if [[ -e "${EROOT%/}${REDMINE_DIR}/config/initializers/session_store.rb" \
+	|| -e "${EROOT%/}${REDMINE_DIR}/config/initializers/secret_token.rb" ]]
+	then
+		elog "Execute the following command to upgrade environment:"
+		elog
+		elog "# emerge --config \"=${CATEGORY}/${PF}\""
+		elog
+		elog "For upgrade instructions take a look at:"
+		elog "http://www.redmine.org/wiki/redmine/RedmineUpgrade"
+	else
+		elog "Execute the following command to initialize environment:"
+		elog
+		elog "# cd ${EROOT%/}${REDMINE_DIR}"
+		elog "# cp config/database.yml.example config/database.yml"
+		elog "# \${EDITOR} config/database.yml"
+		elog "# chown redmine:redmine config/database.yml"
+		elog "# emerge --config \"=${CATEGORY}/${PF}\""
+		elog
+		elog "Installation notes are at official site"
+		elog "http://www.redmine.org/wiki/redmine/RedmineInstall"
+	fi
+}
+
+pkg_config() {
+	# Remove old lock file
+	rm -fr "${EROOT%/}${REDMINE_DIR}/Gemfile.lock"
```

Comment:
Why `-r`, is this a directory? 

Pretty much all of these comments apply to 2nd ebuild as well.